js_hw_7/task_02.js

```diff
@@ -0,0 +1,146 @@
+//Copied from the previous task
+class Animal {
+    constructor(type, color, weight, height, placeOfOrigin) {
+        this.type = type;
+        this.color = color;
+        this.weight = weight;
+        this.height = height;
+        this.placeOfOrigin = placeOfOrigin;
+    };
+    getInfo() {
+        return `Type: ${this.type} color: ${this.color} weight: ${this.weight} height: ${this.height} placeOfOrigin: ${this.placeOfOrigin}`
+    };
+
+    get color() {
+            return this._color;
+          };
+
+    set color(newColor) {
+        if (newColor !== 'Красный' && newColor !== 'Черный' && newColor !== 'Белый' && newColor !== 'Синий') throw new Error ('Invalid color');
+        this._color = newColor;
+    };      
+};
+
+class Snake extends Animal {
+    #isPoisonous
+    constructor (type, color, weight, height, placeOfOrigin, isPoisonous) {
+        super (type, color, weight, height, placeOfOrigin);
+        this.#isPoisonous = isPoisonous;
+    };
+    checkPoisonous(){
+       if (!(typeof this.#isPoisonous === 'boolean')) throw new Error ('isPoisonous should be boolean value');
```

Comment:
This check should be performed in counstructor when setting isPoisonous value

Reply:
Done